internal/backend/runtime/omni/controllers/omni/machine_status.go

```diff
@@ -417,7 +417,8 @@ func (ctrl *MachineStatusController) handleNotification(ctx context.Context, r c
 			spec.PlatformMetadata = event.PlatformMetadata
 		}
 
-		if event.TalosMachineStatus != nil {
+		if event.TalosMachineStatus != nil && spec.TalosMachineStatus != nil &&
```

Comment:
hm... if spec.TalosMachineStatus is nil (initially), how it would be set for the first time?

Reply:
oh, good catch